File: TPBoardWebApi/Services/TableElementService.cs
using Microsoft.AspNetCore.Mvc;
using TPBoardWebApi.Interfaces;
using TPBoardWebApi.Models;

namespace TPBoardWebApi.Services
{
    public class TableElementService : ITableElementService
    {
        private readonly IUnitOfWork _unitOfWork;

        public TableElementService(IUnitOfWork unitOfWork)
        {
            _unitOfWork = unitOfWork;
        }

        public IEnumerable<TableElement> GetAllTableElements()
        {
            return _unitOfWork.TableElements.GetAll();
        }

        public TableElement GetTableElementById(int id)
        {
            return _unitOfWork.TableElements.GetById(id);
        }

        public void CreateTableElement(TableElement element)
        {
            _unitOfWork.TableElements.Add(element);
            _unitOfWork.Save();
        }

        public void UpdateTableElement(TableElement element)
        {
            _unitOfWork.TableElements.Update(element);
            _unitOfWork.Save();
        }

        public void DeleteTableElement(int id)
        {
            var element = _unitOfWork.TableElements.GetById(id);
            if (element != null)
            {
                _unitOfWork.TableElements.Delete(element);
                _unitOfWork.Save();
            }
        }
        public IEnumerable<TableElement> GetTableElementsByTableId(int tableId)
        {
            return _unitOfWork.TableElements.Find(te => te.TableId == tableId).ToList();
        }

        public void AssignUserToTableElement(int tableElementId, int userId)
        {
            var tableElement = _unitOfWork.TableElements.GetById(tableElementId);
            if (tableElement == null) throw new Exception("TableElement not found");

            var user = _unitOfWork.Users.GetById(userId);
            if (user == null) throw new Exception("User not found");

            tableElement.AssignedUserId = userId;
            _unitOfWork.Save();
        }
    }
}

File: TPBoardAngular/src/app/services/table-element.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { TableElement } from '../interfaces/tableelement.model';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class TableElementService {
  private apiUrl = 'https://localhost:7134/api/TableElement';

  constructor(private http: HttpClient) { }

  getAllTableElements(): Observable<TableElement[]> {
    return this.http.get<TableElement[]>(`${this.apiUrl}/GetAllTableElements`);
  }

  getTableElementById(id: number): Observable<TableElement> {
    return this.http.get<TableElement>(`${this.apiUrl}/GetTableElementById/${id}`);
  }

  getTableElementsByTableId(tableId: number): Observable<TableElement[]> {
    return this.http.get<TableElement[]>(`${this.apiUrl}/GetElementsByTableId/${tableId}`);
  }

  createTableElement(tableElement: TableElement): Observable<TableElement> {
    return this.http.post<TableElement>(`${this.apiUrl}/CreateTableElement`, tableElement);
  }

  updateTableElement(tableElement: TableElement): Observable<TableElement> {
    return this.http.put<TableElement>(`${this.apiUrl}/UpdateTableElement/${tableElement.id}`, tableElement);
  }  

  deleteTableElement(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/DeleteTableElement/${id}`);
  }
  assignUserToTableElement(tableElementId: number, userId: number | null): Observable<void> {
    return this.http.post<void>(`${this.apiUrl}/${tableElementId}/assign/${userId}`, {});
  }
}
